aggregate/billing_functions/utils.py

```diff
@@ -0,0 +1,907 @@
+            if not b['time_completed'] or not b['complete']:
+                skipped += 1
+                continue
+
+            time_completed = parse_hail_time(b['time_completed'])
+            in_date_range = start <= time_completed < end
+
+            if time_completed < start:
+                logger.info(
+                    f'{billing_project} :: Got {len(batches)} batches '
+                    f'in {n_requests} requests, skipping {skipped}'
+                )
+                return batches
+            if in_date_range:
+                batches.append(b)
+            else:
+                skipped += 1
+
+
+async def get_jobs_for_batch(batch_id, token: str) -> list[dict[str, any]]:
+    """
+    For a single batch, fill in the 'jobs' field.
+    """
+    jobs = []
+    last_job_id = None
+    end = False
+    iterations = 0
+
+    async with aiohttp.ClientSession() as session:
+        while not end:
+            iterations += 1
+
+            if iterations > 1 and iterations % 5 == 0:
+                logger.info(f'On {iterations} iteration to load jobs for {batch_id}')
+
+            q = '?limit=9999'
+            if last_job_id:
+                q += f'&last_job_id={last_job_id}'
+            url = HAIL_JOBS_API.format(batch_id=batch_id) + q
+
+            jresponse = await async_retry_transient_get_json_request(
+                url,
+                aiohttp.ClientError,
+                session=session,
+                headers={'Authorization': 'Bearer ' + token},
+            )
+            new_last_job_id = jresponse.get('last_job_id')
+            if new_last_job_id is None:
+                end = True
+            elif last_job_id and new_last_job_id <= last_job_id:
+                raise ValueError('Something fishy with last job id')
+            last_job_id = new_last_job_id
+            jobs.extend(jresponse['jobs'])
+
+    return jobs
+
+
+async def migrate_entries_from_hail_in_chunks(
+    start: datetime,
+    end: datetime,
+    func_get_finalised_entries_for_batch,
+    billing_project: str = None,
+    entry_chunk_size=500,
+    batch_group_chunk_size=30,
+    log_prefix: str = '',
+    dry_run=False,
+) -> int:
+    """
+    Migrate all the seqr entries from hail batch,
+    and insert them into the aggregate table.
+
+    Break them down by dataset, and then proportion the rest of the costs.
+    """
+
+    # pylint: disable=too-many-locals
+    token = get_hail_token()
+    result = 0
+    lp = f'{log_prefix} ::' if log_prefix else ''
+
+    batches = await get_finished_batches_for_date(
+        start=start, end=end, token=token, billing_project=billing_project
+    )
+    if len(batches) == 0:
+        return 0
+
+    chnk_counter = 0
+    nchnks = math.ceil(len(batches) / entry_chunk_size) * batch_group_chunk_size
+
+    # Process chunks of batches to avoid loading too many entries into memory
+    for btch_grp in chunk(batches, entry_chunk_size):
```

Comment:
Removing a few characters from the variable names doesn't make them much shorter, but much less readable.

How about `chunk_counter` / `batch_group` instead of `chnk_counter` / `btch_grp`?